bulk_grades/api.py

```diff
@@ -359,37 +359,45 @@ def preprocess_row(self, row):
         operation = {}
         if row['user_id'] in self._users_seen:
             return operation
+
+        operation['new_override_grades'] = []
+        operation['course_id'] = self.course_id
+        operation['user_id'] = row['user_id']
+
         for key in row:
             if key.startswith('new_override-'):
                 value = row[key].strip()
                 if value:
                     short_id = key.split('-', 1)[1]
                     subsection = self._subsections[short_id][0]
-                    operation['user_id'] = row['user_id']
-                    operation['course_id'] = self.course_id
-                    operation['block_id'] = str(subsection.location)
+                    block_id = str(subsection.location)
                     try:
-                        operation['new_override'] = float(value)
+                        new_grade = float(value)
                     except ValueError as error:
                         raise ValidationError(_('Grade must be a number')) from error
-                    if operation['new_override'] < 0:
-                        raise ValidationError(_('Grade must be positive'))
+                    else:
+                        if new_grade < 0:
+                            raise ValidationError(_('Grade must be positive'))
```

Comment:
Nit/pedantic: It's a carry over from the original code, but the test and the message don't agree. Zero isn't a positive value, but an allowed one.